pkg/controller/controller.go

```diff
@@ -97,16 +97,39 @@ func (c *Controller) initOperatorConfig() {
 		c.logger.Infoln("no ConfigMap specified. Loading default values")
 	}
 
-	// env var takes priority over the same param from the operator ConfigMap
-	watchedNamespace := os.Getenv("WATCHED_NAMESPACE")
-	if watchedNamespace != "" {
-		c.logger.Infof("Watch the %q namespace specified in the env variable WATCHED_NAMESPACE\n", watchedNamespace)
-		configMapData["watched_namespace"] = watchedNamespace
+	watchedNsConfigMapVar, isPresentInOperatorConfigMap := configMapData["watched_namespace"]
+	watchedNsEnvVar, isPresentInOperatorEnv := os.LookupEnv("WATCHED_NAMESPACE")
+
+	if (!isPresentInOperatorConfigMap) && (!isPresentInOperatorEnv) {
+		c.logger.Infoln("Neither the operator config map nor operator pod's environment define a namespace to watch. Default to watching all namespaces.")
+		configMapData["watched_namespace"] = v1.NamespaceAll
 	}
 
-	if configMapData["watched_namespace"] == "" {
-		c.logger.Infoln("No namespace to watch specified. Fall back to watching the 'default' namespace.")
-		configMapData["watched_namespace"] = v1.NamespaceDefault
+	if (isPresentInOperatorConfigMap) && (!isPresentInOperatorEnv) {
+
+		// special case: v1.NamespaceAll currently also evaluates to the empty string
+		// so when the param evaluates to the empty string, we use the default ns
+		// since the meaning of the param is a single namespace and *not* all namespaces
+		if watchedNsConfigMapVar == "" {
+			c.logger.Infof("The watched namespace field in the operator config map evaluates to the empty string, falling back to watching the 'default' namespace.\n")
```

Comment:
I still don't think this is quite ideal :( Having the meaning of this variable tied to specifying one individual namespace has a couple of consequences that I would say are not too desirable:

1. No options beyond "one" or "all": If this configuration variable means specifically *one* namespace to watch, then it would be difficult to adapt this to use cases like smoke-testing the multiple namespaces feature if the tests run on a cluster that also has a non-testing operator running.  If this supported listing, say, 2 or 3 namespaces (comma-separated or somesuch) instead of just one (or "all" when absent), then you could have the smoke tests specify 2 or 3 namespaces to listen to, all whose names would include the random test namespace string.  That way, the test operator would not interfere with the non-test operator.  This would not completely fix the issue of such tests, because if the non-test operator listens on *all* namespaces anyway, then the non-test operator would interfere with the test operator; therefore it's possible this is a misplaced concern, and this needs a broader solution (maybe a pattern for what namespaces should be watched, or a pattern for excluded namespaces, or somesuch, although that would bring additional complexity, ugh).

2. No complete environment override: The current semantics mean that if the ConfigMap specifies a `watched_namespace`, even if blank (which would default to the `default` namespace), then it is impossible to override this behavior from an environment variable to make the operator listen to all namespaces.  This is not a big problem as the real benefit of the environment variable override is to get a specific namespace name from `valueFrom` in Kubernetes deployments, and if you want the Operator to listen on all namespaces and you control its environment, then you can (and must) also easily control the contents of the ConfigMap.  Still, to me it feels like it should be possible to override this anyway.

3. Too implicit: I agree with @erthalion that toggling nontrivial behavior like "please listen on all namespaces" through a very implicit mechanism like the *absence* of configuration keys is not ideal and may be surprising for users.  It may be better to have something explicit — it's fine for the default behavior to be "all namespaces", but it seems to me that it should at least deserve a way to explicitly ask for it by setting a key to some value, not just leaving it unset.  Having a separate key or an explicit `*` name/pattern or somesuch would also avoid the inability to override noted in (2).

All of these issues are possibly a matter of preference and/or small importance, so maybe this PR is fine as it is right now.  I still think it's worth mentioning so we explicitly acknowledge this is the design we want despite these issues.

Reply:
> No options beyond "one" or "all": 

I feel watching a few namespaces specified beforehand is beyond the scope of this issue. Let's finish the simpler use case first and extend it later if needed.

>  an absent key meaning to listen only on the default namespace, and a present key set to the empty string (in the ConfigMap or the environment, with the environment taking precedence) meaning listening on all namespaces. 

Makes perfect sense, will have a look into that.

Reply:
Totally agree on limiting scope; I think that's a separate feature (but you will need it as soon as this lands as your smoke tests will break ;) )

Reply:
@mgomezch watching on multiple namespaces has its technical issues: you can easily create a watcher not tied to any namespaces or tied to a single one. For multiple ones, you need multiple watchers. That would add some complexity to the code, and the benefits are not clear to me, as our testing cases are completely covered with running the operator in a single namespace (in the test cluster, we should avoid running any operators that listen to all namespaces, as we don't own that cluster and other people may start experimenting with it).